codegenerator/cli/src/config_parsing/human_config.rs

```diff
@@ -93,6 +93,11 @@ pub struct NetworkContract<T> {
     pub config: Option<T>,
 }
 
+#[derive(Deserialize)]
+pub struct ConfigDiscriminant {
+    pub ecosystem: Option<String>,
```

Comment:
Why not make this an enum?

Reply:
I wanted to handle the invalid ecosystem error nicely, instead of the plain serde one